test/metabase/api/geojson_test.clj

```diff
@@ -140,7 +143,27 @@
              (mt/user-http-request :crowberto :get 200 "geojson" :url test-geojson-url))))
     (testing "error is returned if URL connection fails"
       (is (= "GeoJSON URL failed to load"
-             (mt/user-http-request :crowberto :get 400 "geojson" :url test-broken-geojson-url))))
+             (mt/user-http-request :crowberto :get 400 "geojson"
+                                   :url test-broken-geojson-url))))
+    (testing "error is returned if URL server never responds (#28752)"
+      ;; use a webserver which accepts a connection and never responds. The geojson endpoint opens a reader to the url
+      ;; and responds with it. And if there are never any bytes going across, the whole thing just sits there. Our
+      ;; test flakes after 45 seconds with `mt/user-http-request` times out. And presumably other clients have similar
+      ;; issues. This ensures we give a good error message in this case.
+      (let [never-responder    (fn silent-async-handler [_request _respond _raise])
+            server             (ring-jetty/run-jetty never-responder
+                                                     {:join?         false
+                                                      :async?        true
+                                                      :port          0
+                                                      :async-timeout 60000})
+            port               (.. server getURI getPort)
+            never-responds-url (str "http://localhost:" port)]
```

Comment:
I hate to be that guy but if `.getURI` or the other stuff somehow fails then the server is going to leak

It would be super duper nice to not do anything before the `try` just to be super safe.

It looks pretty safe here, but I'm more worried that people will see this code as an example and copy the pattern in the future and do something not so safe. Better if people see that you have the `try-catch` right away. 

Or if you want to be really l33t maybe you could just reify `java.io.Closeable` and have `(close) call `.stop` and then you could use `with-open`. Idk

Reply:
done. good call.